FWCore/ParameterSet/python/Config.py

```diff
@@ -361,56 +386,89 @@ def __setattr__(self,name,value):
                    oldValue = getattr(self,name)
                    if hasattr(oldValue,'_filename'):
                       oldFile = oldValue._filename
-                   msg1 = "Trying to override definition of "+name+" while it is used by the sequence "
+                   msg1 = "Trying to override definition of "+name+" while it is used by the "
                    msg2 = "\n new object defined in: "+newFile
                    msg2 += "\n existing object defined in: "+oldFile
-                   s = self.__findFirstSequenceUsingModule(self.sequences,oldValue)
+                   s = self.__findFirstUsingModule(self.sequences,oldValue)
                    if s is not None:
-                      raise ValueError(msg1+s.label_()+msg2)
-                   s = self.__findFirstSequenceUsingModule(self.paths,oldValue)
+                      raise ValueError(msg1+"sequence "+s.label_()+msg2)
+                   s = self.__findFirstUsingModule(self.paths,oldValue)
                    if s is not None:
-                      raise ValueError(msg1+s.label_()+msg2)
-                   s = self.__findFirstSequenceUsingModule(self.endpaths,oldValue)
+                      raise ValueError(msg1+"path "+s.label_()+msg2)
+                   s = self.__findFirstUsingModule(self.endpaths,oldValue)
                    if s is not None:
-                      raise ValueError(msg1+s.label_()+msg2)
-            self.__delattr__(name)
```

Comment:
When __delattr__ is called from anywhere other than __setattr__, the new code does everything it did before and some new additional things. The additional things are that it removes deleted modules from sequences, paths, endpaths and tasks. This will help with the problem that had been reported before (in #15792) of None's showing up in the output of dumpPython in configurations where a module was deleted. It also sets a flag in a service so the service knows it is no longer attached to the process.

Previously, __setattr__ called __delattr__, but now instead it calls a new function which does almost exactly the same things except it does not remove the module from sequences, paths, end paths and tasks. The new function is called _delattrFromSetattr. Both this new function and __delattr__ call a new function _delHelper which contains the common parts.